lib/on-contract-event.js

```diff
@@ -0,0 +1,100 @@
+// eth_getFilterChanges isn't fully supported by lotus, see
+// https://filecoinproject.slack.com/archives/CRK2LKYHW/p1705404161018769
+// Therefore we need to poll for events
+
+import { ethers } from 'ethers'
+import timers from 'node:timers/promises'
+
+const MIN_POLL_INTERVAL = 10_000
+
+const makeRequest = async ({
+  contract,
+  address,
+  rpcUrl,
+  rpcHeaders,
+  lastBlockNumber
```

Comment:
In the context of `makeRequest`, `fromBlock` or `fromBlockNumber` would better describe what is this argument used for.

```suggestion
  fromBlock
```

It will also simplify the implementation below.

```diff
- lastBlockNumber: _lastBlockNumber
+ fromBlock: _lastBlockNumber
```

Reply:
nice! while it does make sense to pass `fromBlock` as the input to the function, it doesn't make sense for it to return it, since we're returning a potentially different value - not the `fromBlock` from the query, but the last block seen

Reply:
I've adjusted the query side